geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDensityFilterTest.scala

```diff
@@ -0,0 +1,149 @@
+  var connection: Connection = null
+  var conf: Configuration = cluster.getConfiguration();
+  val TEST_FAMILY = "an_id:java.lang.Integer,attr:java.lang.Double,dtg:Date,geom:Point:srid=4326"
+  val TEST_HINT = new Hints()
+  val typeName = "testpoints"
+
+  step {
+    logger.info("Starting embedded hbase")
+    conf.set(CoprocessorHost.REGION_COPROCESSOR_CONF_KEY,
+      "org.locationtech.geomesa.hbase.coprocessor.KryoLazyDensityCoprocessor");
+    cluster.startMiniCluster(1)
+    connection = cluster.getConnection
+    logger.info("Started")
+  }
+
+  "HBaseDataStore" should {
+    "work with points" in {
+
+      val params = Map(ConnectionParam.getName -> connection, BigTableNameParam.getName -> "test_sft")
+      val ds = DataStoreFinder.getDataStore(params).asInstanceOf[HBaseDataStore]
+
+      ds.getSchema(typeName) must beNull
+
+      ds.createSchema(SimpleFeatureTypes.createType(typeName, TEST_FAMILY))
+
+      val sft = ds.getSchema(typeName)
+
+      sft must not(beNull)
+
+      val fs = ds.getFeatureSource(typeName).asInstanceOf[SimpleFeatureStore]
+
+      val toAdd = (0 until 150).map { i =>
+        val sf = new ScalaSimpleFeature(i.toString, sft)
+        sf.setAttribute(0, i.toString)
+        sf.setAttribute(1, "1.0")
+        sf.setAttribute(2, new DateTime("2012-01-01T19:00:00", DateTimeZone.UTC).toDate)
+        sf.setAttribute(3, "POINT(-77 38)")
+        sf
+      }
+
+      val ids = fs.addFeatures(new ListFeatureCollection(sft, toAdd))
+
+      val table: TableName = connection.getAdmin.listTableNames("test_sft_testpoints_z3")(0)
+
+      val scanRange: Scan = new Scan()
+
+      def rowAndValue(result: Result): RowAndValue = {
+        val cell = result.rawCells()(0)
+        RowAndValue(cell.getRowArray, cell.getRowOffset, cell.getRowLength,
+          cell.getValueArray, cell.getValueOffset, cell.getValueLength)
+      }
+
+      val deserializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId)
+
+      import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+      def getIdFromRow(sft: SimpleFeatureType): (Array[Byte], Int, Int) => String = {
+        val start = if (sft.isTableSharing) {
+          12
+        } else {
+          11
+        } // table sharing + shard + 2 byte short + 8 byte long
+        (row, offset, length) => new String(row, offset + start, length - start, StandardCharsets.UTF_8)
+      }
+
+      val getId = getIdFromRow(sft)
+
+      // 1. Goal scan one of the tables with data
+      val scan = new HBaseBatchScan(connection, table, Seq(scanRange), 2, 100000)
+      val results = scan.map { result =>
+        val RowAndValue(row, rowOffset, rowLength, value, valueOffset, valueLength) = rowAndValue(result)
+        val sf: SimpleFeature = deserializer.deserialize(value, valueOffset, valueLength)
+        sf
+      }
+
+      getDensity("INCLUDE", fs).foreach { i =>
+        logger.info(s"Got $i")
+      }
+
+      true mustEqual (true)
```

Comment:
need some real tests that call this through the regular data store query method, like this: https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DensityIteratorTest.scala#L48-L60